samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/EditNoteActivity.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.firebasecontinue.sample.continote;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.View;
+import android.widget.Button;
+
+import com.firebasecontinue.FirebaseContinue;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseReference;
+
+/**
+ * The Activity that presents the user with a Note editor to edit a specific Note, if
+ * said Note is found for the user.
+ *
+ * TODO: Finish this Activity.
+ */
+public class EditNoteActivity extends BaseActivity {
+
+    // The key from the Firebase Realtime Database of the Note the user wants to edit.
+    // The Note the user wants to edit will be gathered from the Firebase Realtime Database
+    // once the user is confirmed to be signed in below. This is gathered separately on this screen,
+    // rather than passed from the MyNotesActivity, to ensure the user is presented with the
+    // latest value for the Note (at the time of opening this screen).
+    @Nullable
+    private String mDatabaseKey = null;
+
+    // Firebase Realtime Database reference for the current user's Note to edit, based on the key.
+    @Nullable
+    private DatabaseReference mDatabaseRef = null;
+
+    // UI elements
+    @Nullable
+    private Button mContinueWritingElsewhereButton = null;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Intent intent = getIntent();
+        mDatabaseKey = (intent != null) ? intent.getStringExtra("databaseKey") : null;
+        if (mDatabaseKey == null) {
+            // We must have a database key to be on this screen.
+            finish();
+            return;
+        }
+
+        setContentView(R.layout.activity_edit_note);
+
+        // Gather the UI elements for this Activity for future manipulation.
+        mContinueWritingElsewhereButton =
+                (Button) findViewById(R.id.continueWritingElsewhereButton);
+    }
+
+    @Override
+    protected void handleUserSignedIn(@NonNull FirebaseUser user) {
+        super.handleUserSignedIn(user);
+
+        if (user == null) {
+            // This should never happen, but just in case.
```

Comment:
Done, thanks.

Anywhere I had the "This should never/not happen..." disclaimer, I now throw an AssertionError.

It isn't the most graceful way to fail, but these scenarios _really_ should not occur anyways, and if they do at least now they will be painfully obvious to debug.